File: CS/WPFDataGridGettingStartedNETCore/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using WPFDataGridGettingStartedNETCore.Models;

namespace WPFDataGridGettingStartedNETCore {
	public class ViewModel : ViewModelBase {
        NorthwindEntities northwindDBContext;

        public ICollection<Order> Orders {
            get => GetValue<ICollection<Order>>();
            private set => SetValue(value);
        }
        public ICollection<Shipper> Shippers {
            get => GetValue<ICollection<Shipper>>();
            private set => SetValue(value);
        }

        public ViewModel() {
            northwindDBContext = new NorthwindEntities();

            northwindDBContext.Orders.Load();
            Orders = northwindDBContext.Orders.Local;

            northwindDBContext.Shippers.Load();
            Shippers = northwindDBContext.Shippers.Local;
        }

        [Command]
        public void ValidateAndSave(RowValidationArgs args) {
            northwindDBContext.SaveChanges();
        }
    }
}

File: VB/WPFDataGridGettingStartedNETCore/ViewModel.vb
Imports System.Collections.ObjectModel
Imports Microsoft.EntityFrameworkCore
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.Native
Imports WPFDataGridGettingStartedNETCore.Models

Namespace WPFDataGridGettingStartedNETCore

    Public Class ViewModel
        Inherits ViewModelBase

        Private northwindDBContext As NorthwindEntities

        Public Sub New()
            If IsInDesignMode Then
                Orders = New ObservableCollection(Of Order)()
                Shippers = New ObservableCollection(Of Shipper)()
            Else
                northwindDBContext = New NorthwindEntities()
                northwindDBContext.Orders.Load()
                Orders = northwindDBContext.Orders.ToObservableCollection()
                northwindDBContext.Shippers.Load()
                Shippers = northwindDBContext.Shippers.ToObservableCollection()
            End If

            ValidateAndSaveCommand = New DelegateCommand(AddressOf ValidateAndSave)
        End Sub

        Public Property Orders As ObservableCollection(Of Order)
            Get
                Return GetValue(Of ObservableCollection(Of Order))()
            End Get

            Private Set(ByVal value As ObservableCollection(Of Order))
                SetValue(value)
            End Set
        End Property

        Public Property Shippers As ObservableCollection(Of Shipper)
            Get
                Return GetValue(Of ObservableCollection(Of Shipper))()
            End Get

            Private Set(ByVal value As ObservableCollection(Of Shipper))
                SetValue(value)
            End Set
        End Property

        Public ReadOnly Property ValidateAndSaveCommand As DelegateCommand

        Private Sub ValidateAndSave()
            northwindDBContext.SaveChanges()
        End Sub
    End Class
End Namespace
